kore/src/Kore/Builtin/List.hs

```diff
@@ -324,6 +327,27 @@ evalSize resultSort [_list] = do
         & return
 evalSize _ _ = Builtin.wrongArity sizeKey
 
+evalUpdateAll :: Builtin.Function
+evalUpdateAll resultSort [_list1, _ix, _list2] = do
+    _list1 <- expectBuiltinList getKey _list1
+    _list2 <- expectBuiltinList getKey _list2
+    _ix <- fromInteger <$> Int.expectBuiltinInt getKey _ix
+    let len1 = Seq.length _list1
+        len2 = Seq.length _list2
+        result
+            | _ix < 0 = return (Pattern.bottomOf resultSort)
+            | len2 == 0 = returnList resultSort _list1
+            | _ix + len2 > len1 = return (Pattern.bottomOf resultSort)
+            | otherwise =
+                returnList
+                    resultSort
+                    ( Seq.take _ix _list1
+                        <> _list2
+                        <> Seq.drop (_ix + length _list2) _list1
+                    )
```

Comment:
Would using intermediate results declared in a let binding make the a little code clearer here?